### ISSUE TITLE
How to replace post-processed fluorescence channels and add new channels

### ISSUE THREAD

**Issue by jordanyx** [SchmollerLab/Cell_ACDC#66]
I'm interested in replacing or retroactively adding some fluorescence channels into positions that have already been analyzed (channel was previously excluded to save storage space). 

For a given position, what is the easiest way to do this, while preserving the same alignment and cropping so that the same segmentation masks can be used? I know there's a utility that lets one apply alignment to an image but what if I want to do this for every position in a movie without having to do it manually?

**Comment by ElpadoCan** [comment]
Hi Jordan, if I understood correctly you would like to do this starting from the raw microscopy file? Because if that's the case I can code a utility for that. At the moment, while still possible, there is no official way to do that.

**Comment by jordanyx** [comment]
Hi Francesco, that's pretty much what we want to do, yeah. Jacob and I were thinking we could do it ourselves by writing something that reads the align_shift.py file for the crop coordinates to crop the to-be-added channel, then adding that into the metadata, but it would also be nice if such a utility were built into ACDC and capable of doing all positions within the same experiment at once.

**Comment by ElpadoCan** [comment]
I agree it can be very useful. I will do it in the next release, thanks for the idea!

**Comment by ElpadoCan** [comment]
Hi Jordan, unfortunately, I just realised now that it is not possible to apply the same background ROI to new, added channels if the data was originally cropped. This is beacuse when you crop Cell-ACDC removes the saved coordinates of the background ROIs and it saves the actual intensity data from the background ROIs. This means that the exact position of the background ROI is lost after saving cropped data. 

This is done because if Cell-ACDC finds the file with the coordinates of the background ROI it would try to get the background data, but if the data is cropped this would result in an error because the cropped data cannot be indexed with the old background ROI. So my solution to this was to delete the coordinates of the background ROIs to force Cell-ACDC to use the saved background data. 

This is of course not ideal, so I will change the data prep to keep the coordinates of the background ROIs even when you crop, but unfortunately this will be available only in the next release, sorry about that :(

If instead you did not crop the data then it is still possible to apply alignment to the newly added channel.

**Comment by jordanyx** [comment]
Hi Francesco, thanks for the explanation!

In our particular case Jacob and I can work around the loss of the background ROI; we've been using our pixel-dependent flatfield correction in lieu of autoBkgr or dataPrepBkgr, so it actually doesn't affect our data to not have the dataPrepBkgr available. Unfortunately we recently realized we were using a bad flatfield correction (i.e. the subtraction image wasn't reflective of the actual background we saw in our data), which was performed upstream of loading the fluorescence file into the Data Prep module, hence the need for an ability to retroactively add a channel with the same cropping/alignment.

So, it would still be very useful to us if there were a utility that could simply add a new channel and apply the align/shift in an automated fashion across every position in an experiment. I recognize it's a niche case of users making a mistake upstream of the ACDC pipeline, so thanks for helping!
